flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -301,10 +301,14 @@ public boolean databaseExists(String databaseName) throws CatalogException {
 	}
 
 	@Override
-	public void dropDatabase(String name, boolean ignoreIfNotExists) throws DatabaseNotExistException,
-			DatabaseNotEmptyException, CatalogException {
+	public void dropDatabase(String name, boolean ignoreIfNotExists, boolean isRestrict)
+			throws DatabaseNotExistException, DatabaseNotEmptyException, CatalogException {
 		try {
-			client.dropDatabase(name, true, ignoreIfNotExists);
+			if (isRestrict) {
+				client.dropDatabase(name, true, ignoreIfNotExists);
+			} else {
+				client.dropDatabase(name, true, ignoreIfNotExists, true);
+			}
```

Comment:
This might be simplified by just:
client.dropDatabase(name, true, ignoreIfNotExists, !isRestrict);